artichoke-backend/src/extn/core/time/trampoline.rs

```diff
@@ -180,37 +296,46 @@ pub fn to_array(interp: &mut Artichoke, time: Value) -> Result<Value, Error> {
 
 // Math
 
-pub fn plus(interp: &mut Artichoke, time: Value, other: Value) -> Result<Value, Error> {
-    let _ = interp;
-    let _ = time;
-    let _ = other;
-    Err(NotImplementedError::new().into())
+pub fn plus(interp: &mut Artichoke, mut time: Value, mut other: Value) -> Result<Value, Error> {
+    let time = unsafe { Time::unbox_from_value(&mut time, interp)? };
+    if unsafe { Time::unbox_from_value(&mut other, interp) }.is_ok() {
+        Err(TypeError::with_message("time + time?").into())
+    } else if let Ok(other) = implicitly_convert_to_int(interp, other) {
```

Comment:
this makes me realize that our `implicitly_convert_to_xx` helpers don't distinguish between "conversion failed" and "conversion raised an Exception".

No action for now, but I'll see if I can put up a PR.

Reply:
Actually after taking a look, I think in practice, the implicit conversion to integer should be the last branch in the if else chain and errors from it should be propagated with `?`.